### DISCUSSION TITLE
Is it possible to get Response from one channel in another

### DISCUSSION THREAD

**Discussion by kairosmarvin** [nextgenhealthcare/connect#5243] [Q&A] [answered]
Hi guys,

I'm trying to implement a little FileWatcher in Mirth for a usecase.
Right now I'd like to have 1 FileWatcher channel that registers files in a directory and sends a list of filepaths to a FileWorker channel that determines what has to be done for the directory.

My 2 Channels are the following:

1. Mirth FileWatcher:
- Source is a Javascript Reader that routes a message to himself every 5 seconds in order to trigger the Source Transformer.
- In the Source Transformer im preparing the directories
- Destination Javascript Writer is the following:

```
if(!"WATCH" == $('messageType')){
	return;
}

var fileWatcherList = getSysConfig(["FileWatcherRootDirectories"])
var fileUtils = new org.apache.commons.io.FileUtils

for (var rootDirectory in fileWatcherList){
		var file = new java.io.File(fileWatcherList[rootDirectory]+"\\queue")
		var fileList = fileUtils.listFiles(file,null, false)
		var fileListArray = createArrayFromFileList(fileList)
		if(fileListArray.length == 0){
			MirthLog("No files found in directory")
			return;
		}
		var fileWatcherResult = { "project" : rootDirectory, "directory" : fileWatcherList[rootDirectory], "fileList" : fileListArray }
		router.routeMessage("Mirth FileWorker", JSON.stringify(fileWatcherResult))
	}

function createArrayFromFileList(fileList){
	var array = []
	if(undefined == fileList){
		return array
	}
	fileList = fileList.toArray()
	var array = []

	for (file in fileList){
		array.push(fileList[file].toString())
	}
	return array
}
```
![Screenshot 2022-06-14 155231](https://user-images.githubusercontent.com/93179810/173594231-7d3925f3-4661-40a6-8701-edd0f6260af7.png)

I also have a Response where I thought i could grab the Response from the other channel, since I've routed a message to it.

My Destination points to another Channel, my second Channel "Mirth FileWorker"

This channel will handle what has to be done with the files, i.e. send it per REST.
If the operation has been exectued successfully/with error I want to send a response back to the FileWatcher channel in order to move the files from the queue subdirectory to successful or error.

Is that even possible to wait for the second channel Response and use it to cleanup the directories in the first channel ?

**Comment by jonbartels** [comment]
Set "respondFrom" in your second channel. That can either send the destination response as is back to the caller or you can use values from the `responseMap`. Then in the response transformer of your first channel you can make it take any actions you need with that response.

**Reply by pacmano1** [reply]
Why 2 channels?

**Comment by tonygermano** [comment, marked as answer]
[RouteMessage](http://javadocs.mirthcorp.com/connect/4.0.1/user-api/com/mirth/connect/server/userutil/VMRouter.html#routeMessage-java.lang.String-java.lang.String-) returns a [Response](http://javadocs.mirthcorp.com/connect/4.0.1/user-api/com/mirth/connect/userutil/Response.html), but you aren't capturing the return value in your code.

You almost always want to check the `Status` of the `Response` to make sure that the downstream channel isn't stopped and is actually getting your messages, as described in the javadoc for `routeMessage`. It will not throw an exception for you.
